### PR TITLE
Feature/902/161 dpf system test

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Adapt system-test to use Embedded DPF to perform file copy, rather than the legacy `DataStreamPublisher` mechanism.

## Why it does that

Include DPF in system test scope

## Linked Issue(s)

#161 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by cpeeyush — APPROVED** [review]

File: samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java [hunk 1]
```diff
@@ -40,23 +36,23 @@
...
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var vault = context.getService(Vault.class);
-        dataOperatorRegistry.registerStreamPublisher(new FileTransferDataStreamPublisher(context.getMonitor(), dataAddressResolver));
+        var monitor = context.getMonitor();
+
+        var sourceFactory = new FileTransferDataSourceFactory();
+        pipelineService.registerFactory(sourceFactory);
 
-        DataFlowController dataFlowController = new InlineDataFlowController(vault, context.getMonitor(), dataOperatorRegistry, dataAddressResolver);
```
Comment (marcgs) [inline samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:58]: When are we going to remove the `InlineDataFlowController`? In the subsequent story when we update the sample?

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    protected TransferResult transferParts(List<DataSource.Part> parts) {
+        for (DataSource.Part part : parts) {
+            String fileName = part.name();
+            try (var input = part.openStream()) {
+                try (var output = new FileOutputStream(file)) {
+                    try {
+                        input.transferTo(output);
+                    } catch (Exception e) {
+                        return getTransferResult(e, "Error transferring file for %s on %s", fileName, null);
```
Comment (marcgs) [inline system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java:41]: Why do we need the last null argument?

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+import static java.lang.String.format;
+import static org.eclipse.dataspaceconnector.spi.response.ResponseStatus.ERROR_RETRY;
+
+class FileTransferDataSink extends ParallelSink {
+    private File file;
+
+    @Override
+    protected TransferResult transferParts(List<DataSource.Part> parts) {
+        for (DataSource.Part part : parts) {
```
Comment (marcgs) [inline system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java:34]: If I understand correctly, currently only 1 part is supported (as there is only one output file). Should we check this here and process the single part instead?

File: samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java (hunk 1 above)
Comment (cpeeyush) [reply samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:58]: It is still used under Sample05. May be we can mark `InlineDataFlowController` as @Deprecated, what do you think @algattik @marcgs ?

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java (hunk 2 above)
Comment (cpeeyush) [reply system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java]: Updated. Thanks.

File: system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java (hunk 3 above)
Comment (algattik) [reply system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java:34]: We could, but this hardly seems like a responsibility for the Sink, given its contract?

File: samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java (hunk 1 above)
Comment (marcgs) [reply samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:58]: Sound good to me. Perhaps with a reference to the issue that will do the work?

**Review by marcgs — APPROVED** [review]

File: samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java (hunk 1 above)
Comment (cpeeyush) [reply samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:58]: I will create an upstream issue after opening these changes for EDC review.

Comment (cpeeyush) [reply samples/04.0-file-transfer/transfer-file/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferExtension.java:58]: created https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1061

**Pull request merged**